src/Browser/Json.php

```diff
@@ -17,6 +19,20 @@ public function __construct(string $source)
         $this->source = $source;
     }
 
+    /**
+     * @param array<mixed> $arguments
+     */
+    public function __call(string $methodName, array $arguments): self
+    {
+        if (!\method_exists(Expectation::class, $methodName)) {
+            throw new \BadMethodCallException("{$methodName} does not exist");
+        }
+
+        Assert::that($this->decoded())->{$methodName}(...$arguments);
+
+        return $this;
+    }
+
```

Comment:
Let's add a named constructor to reduce duplication.

```suggestion

/**
 * @param mixed $data
 */
public static function encode($data): self
{
    return new self(\json_encode($data, \JSON_THROW_ON_ERROR));
}

```